content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+```
+
+"See this step in Github":https://github.com/ably/tutorials
+
+h2. Step 3 - Add Ably publish and subscribe
+Firstly, add this import statement to @index.js@ file to enable use of @semantic-ui-css@ :
+
+```[javascript]
+import 'semantic-ui-css/semantic.min.css'
+```
+
+Now, open empty the file @App.css@ and add the following:
+
+```[stylesheet]
+.App-header {
+  margin-top: 10px
+}
+
+.App-button {
+  margin-top: 5px;
+  float: right;
+}
+
+.App-users {
+  margin-top: 10px;
+  text-align: left
+}
+
+.App-update {
+  margin-bottom: 10px;
+  cursor: pointer;
+  text-align: center
+}
+
+.App-cards {
+  overflow-y: scroll;
+  max-height: 500px
+}
+```
+
+Now, empty the file @App.js@ then add the following imports:
+
+```[javascript]
+import React, { useState, useEffect } from "react";
+import * as Ably from 'ably';
+import './App.css';
+```
+Now, let's add arrays for avatars and names. A random function also to choose from the arrays:  
+
+```[javascript]
+const avatarsInAssets = [
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_8.png?1536042504672',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_3.png?1536042507202',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_6.png?1536042508902',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_10.png?1536042509036',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_7.png?1536042509659',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_9.png?1536042513205',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_2.png?1536042514285',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_1.png?1536042516362',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_4.png?1536042516573',
+    'https://cdn.glitch.com/0bff6817-d500-425d-953c-6424d752d171%2Favatar_5.png?1536042517889'
+]
+
+const names = ["Ross", "Monica", "Rachel", "Joey", "Chandler", "Steve", "Bill", "Elon", "Tom", "Shaun" ]
+function getRandomArbitrary(min, max) {
+    return Math.floor(Math.random() * (max - min) + min);
+}
+```
+
+Now, let's create the Ably instance and channels. Replace '<YOUR-API-KEY>' with your own API key value. Assign the current user its name and avatar as well:
+
+```[javascript]
+    const myId = "id-" + Math.random().toString(36).substr(2, 16);
+    const apiKey = '<YOUR-API-KEY>';
+    const ably = new Ably.Realtime({
+        key: apiKey,
+        clientId: myId,
+        echoMessages: false
+    });
+    const chatChannel = ably.channels.get("chat");
+    const presenceChannel = ably.channels.get("presence");
+    let my = {};
+    my.avatar = avatarsInAssets[getRandomArbitrary(0, 9)];
+    my.name = names[getRandomArbitrary(0, 9)]
+```
+
+Now, let’s make the functional @App@ component and states within it to hold the user input as well as the updates coming in:
```

Comment:
You've explained what the piece of code does but not how i.e. it'll be helpul if you can add a bit more detailed explanation on what each line of code does (unless is extremely obvious ofc)